include/IVideoDriver.h

```diff
@@ -20,6 +20,11 @@
 
 namespace irr
 {
+namespace asset
```

Comment:
you can just suck in the include files from core, io and asset 

no need for forward declarations

Reply:
the less unnecessary `#include`s in header files, the lesser chance for circular deps

Reply:
well the rule is that core,io and asset shall not include anything from video namespace

`core::smart_refctd_ptr<T>` will break if T is forward declared but not defined.